src/services/localSSHService.ts

```diff
@@ -44,12 +48,32 @@ export class LocalSSHService extends Disposable implements ILocalSSHService {
                     // event, so ignore it if more settings are affected at the same time.
                     return;
                 }
-                if (this.initialized) {
-                    await this.initialized;
-                }
-                this.initialized = this.initialize();
+                await this.initWhenFlagEnabled();
             }
         }));
+        this.experimentService.onDidChangeFlag(async () => {
+            await this.initWhenFlagEnabled();
+        });
+    }
+
+    public async tryInitialize() {
+        if (this.initialized) {
+            return this.initialized;
+        }
+        this.initialized = this.initialize();
+    }
+
+    private async initWhenFlagEnabled() {
+        await this.sessionService.signIn(this.hostService.gitpodHost);
```

Comment:
Configcat only works when we provide userId [[1](https://github.com/gitpod-io/gitpod-vscode-desktop/pull/69/files#r1205711127)] in most cases. (Only when it enabled/disabled all users not need it)